security-proxy/src/main/java/org/georchestra/security/HeadersManagementStrategy.java

```diff
@@ -303,59 +313,114 @@ public synchronized void copyResponseHeaders(HttpServletRequest originalRequest,
-                    continue;
+            logger.debug("Parsing header: \"{}\"", header.toString());
+            List<HttpCookie> currentCookies = HttpCookie.parse(header.getValue());
+            // Normally, we should be only interested in the first element of the list.
+            if (currentCookies.isEmpty()) {
+                continue;
+            }
+            HttpCookie currentCookie = currentCookies.get(0);
+            // if the cookie is a JSESSIONID, we need to store it in the session map,
+            // but do not transmit it to the client, as it could disrupt the current
+            // session between the user and the SP (if no path set, for instance).
+            if (currentCookie.getName().equalsIgnoreCase(HeaderNames.JSESSION_ID)) {
+                // Do not store the JSESSIONID in session if no path is provided
+                // Note: by default, Java webapps seem to limit to their own scope.
+                if (currentCookie.getPath() != null) {
+                    logger.debug("Storing the JSESSIONID into session for path {}", currentCookie.getPath());
+                    storeJsessionHeader(session, currentCookie.getPath(), currentCookie.toString());
                 }
-                if (cookie.trim().startsWith(HeaderNames.JSESSION_ID)) {
-                    String path = "";
-                    if (parts.length == 2) {
-                        path = parts[1];
+            }
+            // Else, it has to be transmitted to the client
+            else {
+                String actualHeaderValue = header.getValue().trim();
+                // if path is not set on the cookie and the overridden path does not point to /
+                if (currentCookie.getPath() == null) {
+                    if (overridenPath != null) {
+                        logger.debug("Current cookie has not path set, forcing it to {}", overridenPath);
+                        if (actualHeaderValue.endsWith(";")) {
+                            actualHeaderValue.concat("Path=/" + overridenPath + ";");
+                        } else {
+                            actualHeaderValue.concat(";Path=/" + overridenPath + ";");
+                        }
                     }
-                    storeJsessionHeader(session, path.trim(), cookie);
-                } else {
-                    if (cookies.length() > 0)
-                        cookies.append("; ");
-                    cookies.append(cookie);
                 }
+                // a path is already set on the cookie, but we make sure to limit its scope
+                // to the current proxified webapp
+                else {
+                    logger.debug("Current cookie {} has a path set to {}, forcing it to {}", currentCookie.getName(),
+                            currentCookie.getPath(), overridenPath);
+                    actualHeaderValue = actualHeaderValue.replaceAll("Path=[^;$]+", "Path=/" + overridenPath);
```

Comment:
`overridenPath` could be `null` here, what we want is maybe an empty string in this case ?

Reply:
if we throw an exception before, we will be ensured that overridenPath won't be null